### PR TITLE
Dev 

### PR DESCRIPTION
**Pull request by roeebenezra**

### CODE REVIEW

File: Controller.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+#include "ImageProcessing.h"
+
+class Controller {
+public:
+    Controller() = default;
+
+    void run();
+
+    void startTest();
```
Comment (excellenteam-scaleup) [inline Controller.h:17]: Should be private if it's only used by run

File: ImageProcessing.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    //    Load the Image and Preprocess
+    cv::Mat image = cv::imread("skin_image.png");
+    cv::Mat gray;
+    cv::cvtColor(image, gray, cv::COLOR_BGR2GRAY);
+    cv::GaussianBlur(gray, gray, cv::Size(5, 5), 0);
+
+    //    Segmentation using Adaptive Thresholding
+    cv::Mat thresh;
+    cv::adaptiveThreshold(gray, thresh, 255, cv::ADAPTIVE_THRESH_MEAN_C, cv::THRESH_BINARY_INV, 11, 2);
```
Comment (excellenteam-scaleup) [inline ImageProcessing.cpp:16]: Document why did you choose these numbers? 255, 11, 2.
Put yourself in the shoes of someone who comes in half a year to fix a bug here...

File: ImageProcessing.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+            // Save or process the cropped image
+            std::string croppedImageName = "mole_" + std::to_string(moleCounter) + ".png";
+            cv::imwrite(croppedImageName, croppedMole);
+            moleCounter++;
+        }
+    }
+
+    // display results
+    cv::imshow("Detected Moles", image);
```
Comment (excellenteam-scaleup) [inline ImageProcessing.cpp:52]: How will this work on Raspberry? Remember, it shouldn't have a GUI...

File: ImageProcessing.cpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    for (const auto &contour: contours) {
+        double area = cv::contourArea(contour);
+        if (area > 500 && area < 1000) {  // These values can be adjusted based on your dataset
+            cv::Point2f center;
+            float radius;
+            cv::minEnclosingCircle(contour, center, radius);
+            cv::circle(image, center, static_cast<int>(radius), cv::Scalar(0, 255, 0), 2);
+
+            // Adjust the cropping region with padding
+            cv::Rect cropRect(center.x - paddingFactor * radius, center.y - paddingFactor * radius,
```
Comment (excellenteam-scaleup) [inline ImageProcessing.cpp:36]: Same comment here, document why you chose these measures

File: ImageProcessing.cpp [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    //    Morphological Operations
+    cv::Mat kernel = cv::getStructuringElement(cv::MORPH_RECT, cv::Size(3, 3));
+    cv::morphologyEx(thresh, thresh, cv::MORPH_OPEN, kernel);
+
+    //     Detect contours
+    std::vector<std::vector<cv::Point>> contours;
+    cv::findContours(thresh, contours, cv::RETR_EXTERNAL, cv::CHAIN_APPROX_SIMPLE);
+
+    //    Filter and Highlight Detected Moles
+    for (const auto &contour: contours) {
```
Comment (excellenteam-scaleup) [inline ImageProcessing.cpp:27]: Function is pretty long, consider dividing this function to several functions?

File: Controller.cpp [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        if (result != 0) {
+            std::cerr << "Error capturing video." << std::endl;
+            continue;
+        }
+
+        // Use FFmpeg to extract image frames from the video
+        std::string framesDirectory = saveDirectory + "/frames_" + std::to_string(currentTime);
+        mkdir(framesDirectory.c_str(), 0777);
+
+        std::string extractFramesCommand = "ffmpeg -i " + videoFilename + " -vf fps=1 " + framesDirectory + "/frame_%d.jpg";
```
Comment (excellenteam-scaleup) [inline Controller.cpp:42]: Use consts

File: ImageProcessing.cpp (hunk 3 above)
Comment (roeebenezra) [reply ImageProcessing.cpp:52]: its not for the raspberry, its for us for now to see the results.

Comment (excellenteam-scaleup) [reply ImageProcessing.cpp:52]: ok so don't forget to remove it...

**Pull request closed without merging**